docs/proposal/kmesh_support_workload.md

```diff
@@ -0,0 +1,141 @@
+  // The services for which this workload is an endpoint. the key is the NamespacedHostname string of the format namespace/hostname.
+  Services map[string]*PortList `protobuf:"bytes,22,rep,name=services,proto3" json:"services,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
+  // Health status of the workload，such as: "Healthy"，"Unhealthy"
+  Status WorkloadStatus `protobuf:"varint,17,opt,name=status,proto3,enum=istio.workload.WorkloadStatus" json:"status,omitempty"`
+}
+```
+
+## How kmesh subscribe and use the workload model info
+
+1. kmesh's xds client subscribes to the workload model from Istiod through Delta manner, and the type_url is: "type.googleapis.com/istio.workload.Address";
+2. The workload data of Address type is divided into two sub resources: Service and Workload, which are responsed to kmesh and then parsed and converted into internal structures, stored in the bpf map;
+
+![kmesh subscribe workload](./kmesh_subscribe_workload.png)
+
+In the subsequent traffic management of kmesh, based on the IP and Port accessed by the client, find the corresponding service and its endpoints from the BPF map, and then randomly select an endpoint to forward the request to that endpoint;
+
+## Kmesh BPF map data structure definition
+
+```C
+// service_info map
+typedef struct
+{
+    __be32 ipv4; // service ip
+    __be16 service_port;
+} __attribute__((packed)) service_info_key;
+
+typedef struct
+{
+    __u32 service_id; // service id, through <namespace>/<hostname> string convert to uint32 variable
+} __attribute__((packed)) service_info_value;
+
+// service map
+typedef struct
+{
+    __u32 service_id; // service id, through <namespace>/<hostname> string convert to uint32 variable
+} __attribute__((packed)) service_key;
+
+typedef struct
+{
+	__u32 endpoint_count; // the endpoint count of the service
+} __attribute__((packed)) service_value;
+
+// endpoint map
+typedef struct
+{
+    __u32 service_id; // service id, through <namespace>/<hostname> string convert to uint32 variable
+    __u32 backend_index; // backend index，The relationship of backend_index and endpoint_count：if endpoint_count is 3，then backend_index can be 0,1,2;
+} __attribute__((packed)) endpoint_key;
+
+typedef struct
+{
+    __u32 backend_uid; // backend uid, through workload_uid string convert to uint32 variable
+} __attribute__((packed)) endpoint_value;
+
+// backend map
+typedef struct
+{
+    __u32 backend_uid; // backend uid, through workload_uid string convert to uint32 variable
+} __attribute__((packed)) backend_key;
+
+typedef struct {
+    __be16 service_port;
+    __be16 target_port;
+} __attribute__((packed)) port_pair;
+
+typedef struct
+{
+    __be32 ipv4; // backend ip
+    __u32 port_count;
+    port_pair portlist[MAX_COUNT];
+} __attribute__((packed)) backend_value;
+
+```
+
+## Subscription data process flow
+
+![workload info operation](./workload_info_operation.png)
```

Comment:
In addition to the design diagrams, please provide a detailed description of the processing logic for each step in the entire workflow, including any potential constraints and limitations.